src/pages/permissions/PermissionGroups.tsx

```diff
@@ -0,0 +1,298 @@
+import {
+  Button,
+  EmptyState,
+  Icon,
+  Row,
+  TablePagination,
+  useNotify,
+} from "@canonical/react-components";
+import { useQuery } from "@tanstack/react-query";
+import Loader from "components/Loader";
+import ScrollableTable from "components/ScrollableTable";
+import SelectableMainTable from "components/SelectableMainTable";
+import SelectedTableNotification from "components/SelectedTableNotification";
+import { FC, useState } from "react";
+import { queryKeys } from "util/queryKeys";
+import useSortTableData from "util/useSortTableData";
+import { getIdentitiesForGroup } from "util/permissionIdentities";
+import usePanelParams, { panels } from "util/usePanelParams";
+import CustomLayout from "components/CustomLayout";
+import PageHeader from "components/PageHeader";
+import NotificationRow from "components/NotificationRow";
+import HelpLink from "components/HelpLink";
+import { useDocs } from "context/useDocs";
+import { fetchGroups } from "api/auth-groups";
+import GroupActions from "./actions/GroupActions";
+import CreateGroupPanel from "./panels/CreateGroupPanel";
+import EditGroupPanel from "./panels/EditGroupPanel";
+import PermissionGroupsFilter from "./PermissionGroupsFilter";
+import EditGroupIdentitiesBtn from "./actions/EditGroupIdentitiesBtn";
+import EditGroupIdentitiesPanel from "./panels/EditGroupIdentitiesPanel";
+import { pluralize } from "util/instanceBulkActions";
+import BulkDeleteGroupsBtn from "./actions/BulkDeleteGroupsBtn";
+import EditGroupPermissionsPanel from "./panels/EditGroupPermissionsPanel";
+
+const PermissionGroups: FC = () => {
+  const notify = useNotify();
+  const {
+    data: groups = [],
+    error,
+    isLoading,
+  } = useQuery({
+    queryKey: [queryKeys.authGroups],
+    queryFn: fetchGroups,
+  });
+  const docBaseLink = useDocs();
+  const panelParams = usePanelParams();
+  const [search, setSearch] = useState("");
+  const [selectedGroupNames, setSelectedGroupNames] = useState<string[]>([]);
+
+  if (error) {
+    notify.failure("Loading groups failed", error);
+  }
+
+  const headers = [
+    { content: "Name", className: "name", sortKey: "name" },
+    { content: "Description", sortKey: "description" },
+    {
+      content: "Identities",
+      sortKey: "identities",
+      className: "u-align--right",
+    },
+    {
+      content: "Permissions",
+      sortKey: "permissions",
+      className: "u-align--right",
+    },
+    { "aria-label": "Actions", className: "u-align--right actions" },
+  ];
+
+  const filteredGroups = groups.filter(
+    (group) =>
+      !search ||
+      group.name.toLowerCase().includes(search) ||
+      group.description.toLowerCase().includes(search),
+  );
+
+  const selectedGroups = panelParams.group
+    ? groups.filter((group) => group.name === panelParams.group)
+    : groups.filter((group) => selectedGroupNames.includes(group.name));
```

Comment:
that would make the component easier to manage. Hoever, this approach caters for when the user have selected multiple groups, then open the panel for a single group via the contextual actions, if they close the panel at this point then the originally selected groups will still show. Is this behaviour maybe not necessary?

Reply:
I don't have a strong opinion, but my gut feeling is it is okay to change that. 

I think it is better to have the selection you saw when the panel was open after closing it. It might be confusing to select 3, then open the panel for 1, only see that 1 selected, and on save or close the previous 3 get selected.

Reply:
Okay perfect I will make that change :+1: